### PR TITLE
Preparations for Python 3.5 D-Day.

### PR DESCRIPTION
**Pull request by ludeeus**
## Description:

This is a big one...

Changes:
 - Various tweaks to both python and homeassistant suites.
 - Added a python-migration function
   - This handles checks if migration is needed and notify the user about it.
   - It "stores" the current pyversion in a file `/srv/homeassistant/hassbian/pythonmigration`
   - If the install script for HA is run, the user will not get a choice (this should only be used the first time anyway)
- Added a new `systeminfo` function (Should be it's own PR...I know....)
  - This prints useful system information before running install/upgrade/remove scripts.
  - If there is an update to `hassbian-config` if will show that as well.

image

**Related issue (if applicable):** Fixes #<hassbian-scripts issue number goes here>

## Checklist (Required):
  - [ ] The code change is tested and works locally.
  - [ ] The code is compliant with [Contributing guidelines](https://github.com/home-assistant/hassbian-scripts/blob/master/.github/CONTRIBUTING.md)

### If pertinent:
  - [x] Script has validation check of the job.
  - [ ] Created/Updated documentation at `/docs`

### CODE REVIEW

File: package/opt/hassbian/suites/python.sh [hunk 1]
```diff
@@ -12,64 +12,49 @@ function python-show-copyright-info {
...
-make altinstall
-apt -y autoremove
-cd  || return 1
-rm -r /tmp/Python-"$PYTHONVERSION"
-rm /tmp/Python-"$PYTHONVERSION".tar.xz
-echo "Done"
 
+isinstalled=$(which "python${PYTHONVERSION:0:3}")
+if [ ! -z "${isinstalled}" ]; then
+  currentversion$("python${PYTHONVERSION:0:3} -V")
```
[the remark this answers is not in the document]
Comment (ludeeus) [reply package/opt/hassbian/suites/python.sh]: There is **nothing** more left to quote...

**Review by Landrash — APPROVED** [review]

**Review by Landrash — CHANGES_REQUESTED** [review]
systeminfo is missing form "tab completion"

**Review by Landrash — CHANGES_REQUESTED** [review]
Error on Stretch

> Checking current version...
3.5.3/opt/hassbian/suites/python.sh: line 29: python3.7: command not found
Upgrading to Python 3.7.3

**Review by Landrash — APPROVED** [review]
LGTM. Tested on Stretch and Buster

**Pull request merged**